### ISSUE TITLE
An immediately closed HTML tag does not parse properly

### ISSUE THREAD

**Issue by stephengardner** [cheeriojs/cheerio#1200]
I'm trying to parse some user's input HTML which happens to have an empty `<div></div>`  tag.
When parsing into XML, I get this: `<div/>`

And then when parsing back into HTML, the `<div>` is then a wrapper for the inner content.

Ex:
```
<div></div>
<p>hello world</p>
```

Parsing into XML  
`const xml = $.load('<div></div><p>hello world</p>', {xmlMode: true});`

Parsing back into HTML:   
`const newHtml = xml.html({decodeEntities: true});`

Gives:
```
<div>
<p>hello world</p>
</div>
```

This is an issue because of styling that might happen on the containing div, etc.

I'm not exactly sure if this is a bug or intentional.  I know that `<div></div>` doesn't make much sense, but it's something I'm dealt with.  To clarify; if the div has a space within it, the HTML is parsed as-expected.

**Comment by fb55** [comment]
What version of cheerio are you using?

**Comment by gajus** [comment]
Related:

```js
const $ = cheerio.load('<script id="data-layer">"<br>"</script>', {
  xmlMode: true
});

console.log($.html());

```

Produces:

```
<script id="data-layer">"<br>&quot;</br></script>

```

Expected:

```
<script id="data-layer">"<br>"</script>

```

**Comment by unicron** [comment]
Using xml, this repros with 0.22.0.  We have found the same behavior using `<strong></strong>` which causes odd wrapping behavior as described.

It appears to parse correctly using html parsing.

**Comment by nicksheen** [comment]
fyi unicron's repro above was obtained from the options
```cheerio.load(cachedComponent.content, {
                decodeEntities: false,
                xmlMode: true,
            });```
If we use xmlMode false then we don't have the problem with immediately closed tags, but we need xmlMode true to handle <br/> tags correctly.

**Comment by mmmpop** [comment]
Came to find this thread apparently. My problem lies with parsing Font Awesome tags in a server-rendered document:

`<i class="fa fa-angle-down"></i>`

The parser will then start to incorrectly assume that the sibling element is a child and really all hell breaks loose. I am using the same settings as @nicksheen .

I discovered the issue when trying to insert multiple script tags with an src attribute but no textContent. Stuffing them with an empty object literal fooled the parser into realizing the following script tag was in fact a sibling element and not a child (and a string of all elements in the head for that matter).

I haven't checked to see if I can work around the FA issue in a similar manner but am trying now.

**Comment by rdeelstra** [comment]
This is certainly an issue, example with <script src="/js/script.js"></script> turns this into a <script src="/js/script.js" /> which is proper XML but HTML is not proper XML (for reasons unknown to me about why this ever was accepted in the first place). My chrome browser breaks because of that for some reason. Is there a workaround are should I look for another solution?

**Comment by stephengardner** [comment]
@mmmpop this is not a bug to you?  Confused.  Your workaround is specifically for React.

There have been multiple good examples of this breaking, can we get a fix? :)

**Comment by fb55** [comment]
Fixed by #985
